lib/index.js

```diff
@@ -2,7 +2,94 @@ const crypto = require('crypto');
 
 const ALPHA_NUM = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
 
-const IdGenerator = function(prefixes){
+// Largest integer we can read from a buffer.
+//const RAND_MAX = 0xFFFF_FFFF_FFFF;
+const RAND_MAX = Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+/*
+ Generates an integer in [min, max) range where min is inclusive and max is
+ exclusive.
+
+ Backported from node.js v14 crypto.randomInt,
+ https://nodejs.org/api/crypto.html#crypto_crypto_randomint_min_max_callback
+ This version eplaces native NumberIsSafeInteger 
+ with Number.isSafeInteger and uses locally defined rand_max values
+
+ This function is licensed under the terms of the nodejs license,
```

Comment:
From a licensing POV it is best to keep code from different sources segregated; could you move this into its own file with a license header?

Even though auth0/id-generator is itself OSS I think it is helpful to keep these separate.

Reply:
Might be worth asking our Compliance team.

Reply:
Switched the function to it's own file with an explicit license statement.  I'll reach out to compliance for guidance now